pontoon/sync/tasks.py

```diff
@@ -283,34 +289,13 @@ def sync_translations(
                 if not vcs_project.synced_locales:
                     vcs_project.resources
 
-                if not obsolete_vcs_resources:
-                    # Skip all locales if none of the them has anything to sync
-                    if len(vcs_project.synced_locales) == 0:
-                        if resources_changed:
-                            for l in locales:
-                                update_translated_resources(db_project, vcs_project, l)
-                                update_locale_project_locale_stats(l, db_project)
-                            db_project.aggregate_stats()
-
-                        log.info(
-                            'Skipping syncing translations for project {0}, none of the locales '
-                            'has anything to sync.'.format(db_project.slug)
-                        )
-
-                        for r in repos:
-                            r.set_last_synced_revisions(locales=repo_locales[r.pk])
-                        repo_sync_log.end()
-                        return
-
-                    # Skip locales that have nothing to sync
-                    if vcs_project.synced_locales and locale not in vcs_project.synced_locales:
-                        if resources_changed:
-                            update_translated_resources(db_project, vcs_project, locale)
-                            update_locale_project_locale_stats(locale, db_project)
-                            log.info('Skipping locale `{0}` for project {1}, no changes detected.'
-                                     .format(locale.code, db_project.slug))
-                            log.info(vcs_project.resources)
-                        continue
+                # Skip all locales if none of the them has anything to sync
+                if len(vcs_project.synced_locales) == 0:
+                    break
```

Comment:
And now that we also update stats after the big `for locale in locales` loop (to c), we can simply break instead of return.